vibe_code_apps_20/wattlab/weather/epw.py

```
@@ -332,6 +332,85 @@ def build_amy_epw(
+            except ValueError:
+                continue
+            if first_row is None:
+                first_row = row_d
+            last_row = row_d
+
+    begin = first_row or header_begin
+    end = last_row or header_end
+    if begin is None or end is None:
+        return None
+    full_year = begin.month == 1 and begin.day == 1 and end.month == 12 and end.day == 31
+    # Same calendar span without requiring Jan1–Dec31 when years differ in TMY
```

Ruff (0.15.21)

[warning] Comment contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF003)